include/llvm-dialects/Dialect/Utils.h

```diff
@@ -42,6 +44,30 @@ bool runInstructionVerifier(
 bool runTypeVerifier(llvm::function_ref<bool(llvm::raw_ostream &)> verifier,
                      llvm::Type *type, llvm::raw_ostream *errs = nullptr);
 
+/// Adapt an operand Use* iterator to a Value* iterator.
+template <typename BaseT>
+class op_to_value_iterator_adaptor_impl
+    : public llvm::iterator_adaptor_base<
+          op_to_value_iterator_adaptor_impl<BaseT>, BaseT,
+          typename std::iterator_traits<BaseT>::iterator_category,
+          llvm::Value *> {
+private:
+  using SelfT = op_to_value_iterator_adaptor_impl<BaseT>;
+
+public:
+  op_to_value_iterator_adaptor_impl() = default;
+  template <typename U>
+  op_to_value_iterator_adaptor_impl(U &&u)
+      : SelfT::iterator_adaptor_base(std::forward<U &&>(u)) {}
```

Comment:
Do default copy and move constructor work together with this forwarding one being present, without having explicitly been declared?

Reply:
I think so. At least, LLVM's pointee_iterator is defined using the same pattern.